Digital 1/Assets/Scripts/Player/PlayerShip.cs

```diff
@@ -58,36 +67,50 @@ void HandleInput()
 		rigidbody.velocity = movement * moveSpeed;
 
 		// shooting
-		if (Input.GetButton("Jump") || Input.GetKey("space"))
+		if (Input.GetAxis("Fire1") == 1.0f)
 		{
 			Fire();
 		}
 
 		// changing color
-		if (Input.GetKeyDown(KeyCode.L))
-		{
-			ColorSwitch(ColorState.Red);
-		}
-		if (Input.GetKeyDown(KeyCode.Semicolon))
-		{
-			ColorSwitch(ColorState.Blue);
-		}
-		if (Input.GetKeyDown(KeyCode.Quote))
-		{
-			ColorSwitch(ColorState.Yellow);
-		}
+        if (Input.GetAxis("Mouse ScrollWheel") > 0) 
```

Comment:
int scroll = Input.GetAxis("Mouse ScrollWheel") > 0 ? 1 : -1;
currentState = (currentState + scroll) % 4;
ColorSwitch(currentState);

if possible try to replace 4 with a non-constant value.